### ISSUE TITLE
Latest update v160 makes fancontrol unusable

### ISSUE THREAD

**Issue by DJPhilGames** [Rem0o/FanControl.Releases#1835]
v159 works perfectly besides avast flagging fan control using a vulnerable driver
v160 stops working after setup, whatever the fan speed is going to when the program starts it stays at that speed and doesn't change, no values of fan speeds change while the application is still responsive 
its like this new update has broken some chain that changes fan speeds and detects fan speeds

My resolution:
reverting back to v159 seems to be the only resolution

Hardware specs:

CPU: Intel i7 10700k
GPU: Nvidia RTX 3080
Ram: 32gb 3200mhz ram
Main SSD: Samsung 2tb 980 pro (Fan control running of this SSD)
2nd SSD: Sabrent 2TB Rocket NVMe PCIe M.2
Motherboard: MSI MPG Z490 GAMING PLUS
Fans: Corsair  CO-9050074-WW LL140

**Comment by alexvojacek** [comment]
Same thing happening here.  It also starts opening on the foreground every time I start the PC and since it's stuck on the same speed, it's basically completely broken.

Similar specs than yours.

**Comment by Rem0o** [comment]
Are you able to run the latest nightly of https://github.com/LibreHardwareMonitor/LibreHardwareMonitor? 

[Link](https://nightly.link/LibreHardwareMonitor/LibreHardwareMonitor/workflows/master/master/LibreHardwareMonitor-net472.zip)

**Comment by DJPhilGames** [comment]
Yes update version v0.9.2 of LibreHardwareMonitor seems to be working perfectly fine with version v159 of fan control

**Comment by Rem0o** [comment]
Is it just when opening the instance that started with Windows?  If you close than instance and manually open FanControl, same issue?

**Comment by JasonHorkles** [comment]
It seems to happen only after restarting Windows, and the only way that worked for me to fix it was to force close it via Task Manager and start it up manually (the built-in quit still had the issue when re-opening the app)

**Comment by Azyrion** [comment]
I have the same Issue. The only way to make it usable is to quit via the taskbar icon or taskmanager force quit or using the update.exe that restarts the app.

CPU: Ryzen 7 5800X3D
RAM: 32GB CL16 3600Mhz
GPU: RX 6900 XT
MB: MSI X570 Tomahawk WiFi
SSD: 1TB Samsung 980 Pro

**Comment by Rem0o** [comment]
[FanControl.IPC.zip](https://github.com/Rem0o/FanControl.Releases/files/12052397/FanControl.IPC.zip)

This is a wild guess, but it's the only significant thing that changed since V160.   For this test, I basically compiled a version of this dll where I removed all inter-process communication.  Empty shell.  So clicking on the FanControl icon while an other instance is running will be broken and so on, but it won't affect the rest of the software.   

Swap it in and see if the problem persists. It's a shot in the dark but without being able to reproduce the issue myself, this is a first try.

**Comment by DJPhilGames** [comment]
Yesterday I updated to the latest version forgetting about my not being able to run v160 and it worked for a day but when I logged in today it would not work regardless of how many times I reset it I'm not sure if what you provided fixed the software after updating but here's my process that seemed to make it work:
Download v160 (Net 4_8), add my configuration & Run (wasn't working)
force close the whole application and replace the file you provided & rerun it (Still not working)
Download v160 (Net 7), add my configuration & run (it wasn't working)
force close the whole application and replace the file you provided & run again (starts working)
update to the newest version and still working even after a system reset
I'll update this with a new comment if it stops working again next week as it always seems to work for a day and then stops again

**Comment by Rem0o** [comment]
@DJPhilGames  over 163, put the FanControl.IPC dll on top, run that for a while, and see if it persists.

**Comment by DJPhilGames** [comment]
I'll run it with that .dll on top for a week while keeping on top of updates and see if that fixes it ill comment again in a week to let you know if I've had any issues if I do have any issues I will probably mention it sooner

**Comment by DJPhilGames** [comment]
It's been a week, I had 1 update which was working without replacing IPC for 2 days but then it started not working today until I replaced the ipc.dll again which is a tedious way to use this software but its a bandage I will need to/be able to apply each update to get it working for me
I want to assume there isn't any way to make this less tedious for me?

**Comment by Rem0o** [comment]
@DJPhilGames  that's because the update doesn't have the stripped down IPC dll. 

On an other thread, same issue, I compiled a special version of the exe with extra logs. 
https://github.com/Rem0o/FanControl.Releases/issues/1938#issuecomment-1646683977

Haven't had any response since.  If you are so inclined, replace the V164 exe with that one, start the program, experience the issue, then send me the logs.txt that will be next to the exe.

**Comment by Azyrion** [comment]
I myself didn't have the same issue today! Gonna give an update Friday/Saturday to see if I was just lucky.

**Comment by Azyrion** [comment]
v164 seems to be working just fine for me now. Did nothing but update from v162 that had the dll replaced by the one posted here. 

Ryzen 7 5800X3D
32GB RAM CL16 3600Mhz
RX 6900 XT Toxic LE
